scratch/camera_feed/display_camera_feed.cpp

```diff
@@ -0,0 +1,165 @@
+    if (tjDecompress2(_jpegDecompressor, compressed_image, _jpegSize, buffer, WINDOW_X_LENGTH, pitch, WINDOW_Y_LENGTH, TJPF_RGB, TJFLAG_FASTDCT) == -1)
+        std::cout << "decompressed incorrectly" << std::endl;
+    //free the decompressor.
+    tjDestroy(_jpegDecompressor);
+}
+
+int main(){
+    int pitch = 1920*3;
+    uint32_t offset;
+    //create a buffer for the message struct (declared in loop)
+    unsigned char* temp_compressed_message_buffer = new unsigned char[CAMERA_FRAME_BUFFER_SIZE];
+    //array of buffers to hold the compressed messages
+    BufferItem* frame_buffer = new BufferItem[CAMERA_FRAME_BUFFER_COUNT];
+    //pointer to where we keep the pixel data
+    unsigned char* pixel_location;
+    for(int i = 0; i < CAMERA_FRAME_BUFFER_COUNT; i++){
+        frame_buffer[i].data_sections = new unsigned char[CAMERA_FRAME_BUFFER_SIZE];
+    }
+    uint8_t indx = 0;
+    network::Connection conn;
+    const char* destination_port = "127.0.0.1";
+    //allow error to fade out of scope
+    {
+        network::Error error = network::connect(&conn, destination_port, 45546, 45545) ;
+        if(error == network::Error::OPEN_SOCKET)
+            std::cout << "Open socket error: " << std::endl;
+        else if(error == network::Error::BIND_SOCKET)
+            std::cout << "Binding socket error" << std::endl;
+    }
+    //check for failure
+    if (SDL_Init(SDL_INIT_VIDEO) != 0){
+        std::cout << "SDL_Init Error: " << SDL_GetError() << std::endl;
+        return 1;
+    }
+    //creates window. pramaeters are window title, x start pos, y start pos,
+    //x end pos, and y end pos. We use the dimensions of the window provided above
+    SDL_Window *win = SDL_CreateWindow("Camera feed!", 0, 0, WINDOW_X_LENGTH, WINDOW_Y_LENGTH, SDL_WINDOW_SHOWN);
+    //check for failure
+    if (win == nullptr){
+        std::cout << "SDL_CreateWindow Error: " << SDL_GetError() << std::endl;
+        SDL_Quit();
+        return 1;
+    }
+    /*
+        makes a renderer. args are the window to render to, the index of the 
+        rendering driver, and the rendering flags or'd together
+        SDL ACCELERATED uses hardware acceleration, and SDL PRESENTVSYNC
+        is present is synchronized with the refresh rate
+        Other flags are TARGETTEXTURE -- supports redering to texture
+        and SOFTWARE -- which means the renderer is a software fallback 
+        should the hardware accelerated rendering fail
+    */
+    SDL_Renderer *ren = SDL_CreateRenderer(win,-1,
+        SDL_RENDERER_ACCELERATED   | SDL_RENDERER_PRESENTVSYNC);
+    if (ren == nullptr){
+        SDL_DestroyWindow(win);
+        std::cout << "SDL_CreateRenderer Error: " << SDL_GetError() << std::endl;
+        SDL_Quit();
+        return 1;
+    }
+    SDL_Texture *texture = SDL_CreateTexture(ren, SDL_PIXELFORMAT_RGB24, 
+        SDL_TEXTUREACCESS_STREAMING, 1920, 1080);
+    bool running = true;
+    SDL_Event event;
+    do{
```

Comment:
Why not just use a while(running) loop? Its clearer when we read top down and it would (theoretically) only require one more check of `running`.